wgpu-native/src/command/mod.rs

```diff
@@ -183,10 +188,24 @@ pub fn command_encoder_begin_render_pass(
     let mut extent = None;
     let mut barriers = Vec::new();
 
+    let limits = HUB.adapters.read()[device.adapter_id].physical_device.limits();
+    let samples_count_limit = limits.framebuffer_color_samples_count;
+
     let color_attachments =
         unsafe { slice::from_raw_parts(desc.color_attachments, desc.color_attachments_length) };
     let depth_stencil_attachment = unsafe { desc.depth_stencil_attachment.as_ref() };
 
+    let sample_count = color_attachments.get(0).map(|at| view_guard[at.attachment].samples).unwrap_or(1);
+    assert!(sample_count & samples_count_limit != 0, "Attachment sample_count must be supported by physical device limits");
```

Comment:
why are we using `&` here?

Reply:
Because its a bitmask mask? https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkPhysicalDeviceLimits.html

Reply:
oh wow, this is confusing. Filed https://github.com/gfx-rs/gfx/issues/2859